### PR TITLE
Modified to take custom git arguments

### PR DESCRIPTION
**Pull request by aiampogi**
- made it multitask so user can specify own targets
- custom git submodule update arguments can now be used by defining an
  option called `gitArgs` for your task target
- if `gitArgs` option is not defined, arguments will still fallback to
  the original `"--init --recursive" + merge` one.
- if `gitArgs` option is `null`, no arguments will be used, which will
  be just `git submodule update`

### CODE REVIEW

**Comment by aiampogi** [comment]
Hi @jaubourg I created this PR to resolve https://github.com/TryGhost/Ghost/issues/2503

This is by enabling the user to specify if he wants to have the default argument, his custom argument or none at all.

**Comment by aiampogi** [comment]
Btw, all tests are passing, I also added 2 new tests for the new functionality. :)

**Comment by aiampogi** [comment]
Oh shoot, I forgot to update the README.md.

Anyways, if you like me to update it after accepting this PR, please let me know, otherwise you can just add below.

###Custom Arguments
You can have your own custom `git submodule update` arguments by defining the `gitArgs` option in your task target.
e.g.

``` javascript
grunt.initConfig({
    "update_submodules": {
        default: {
            options: {
                // gitArgs: undefined // really commented to have this undefined, this will do the default git submodule update --init --recursive
            }
        },
        withCustomArgs: {
            options: {
                gitArgs: "--force"
            }
        },
        withBlankArgs: {
            options: {
                gitArgs: null // there will be no arguments for your git submodule update
            }
        }
    }
});
```

File: lib/task.js [hunk 1]
```diff
@@ -6,13 +6,21 @@ module.exports = {
...
+			var params;
+			if (options.gitArgs === undefined) {
+				params = " --init --recursive" + merge;
+			} else if (options.gitArgs === null) {
+				params = "";
+			} else {
+				params = " " + options.gitArgs;
+			}
+
+			runner.exec( "git submodule update" + params, runner.ok );
```
Comment (jaubourg) [inline lib/task.js]: This seems quite convoluted to me. How about something like this;

``` js
var gitArgs = options.hasOwnProperty( "gitArgs" ) ? options.gitArgs || "" : "--init --recursive" + merge;
runner.exec( "git submodule update " + gitArgs );
```

That way:
- if you don't have a `gitArgs` option, default is used,
- if you do have a falsy `gitArgs` option (`false` -- obvious favorite for documentation --, `""`, `null`, `undefined`), pure `git submodule update`
- if you do have a non falsy `gitArgs`, it's used as the command line arguments

What do you think?

Oh, also, why naming this variable `params` when those actually are `gitArgs`?

Comment (aiampogi) [inline lib/task.js]: That is perfectly fine! If you can do a manual merge, I would really
appreciate. I'm currently out of town. I really would appreciate it! :)
thanks!!

**Comment by aiampogi** [comment]
@jaubourg There, I've squashed all commits into one. I've updated the code to make use of your suggestion. :)

**Comment by jaubourg** [comment]
Just waiting on Travis to run tests before releasing on npm.

**Pull request merged**